src/NUnitEngine/nunit.engine/RuntimeFramework.cs

```diff
@@ -108,6 +98,9 @@ public RuntimeFramework(RuntimeType runtime, Version version, string profile)
             Profile = profile;
 
             DisplayName = GetDefaultDisplayName(runtime, FrameworkVersion, profile);
+
+            if (clrVersion != null)
+                ClrVersion = clrVersion;
```

Comment:
This is messy, and solely here to retain previous functionality. ClrVersion is set differently depending on if the framework is found as a "CurrentFramework" or an "AvailableFramework". I'd like to standardise that, but it would be a change to the tests, so I can do that separately later.